src/main/core/scheduler/scheduler.c

```diff
@@ -188,13 +188,13 @@ gboolean scheduler_push(Scheduler* scheduler, Event* event, Host* sender, Host*
     utility_debugAssert(receiver);
 
     /* push to a queue based on the policy */
-    schedulerpolicy_push(
+    eventTime = schedulerpolicy_push(
```

Comment:
Slight preference toward calling `event_getTime` instead of adding the return value to `schedulerpolicy_push`.

Adding it as a return value makes the API seem like the scheduler might schedule it for some other time than the specified event time.

Reply:
I'm not sure what you mean here. The `schedulerpolicy_push()` may change the event time, but it also takes ownership of the event so that we can't access it with `event_getTime` later.

One alternative could be to move the code where `schedulerpolicy_push()` changes the event time into this parent function. Another could be to make the `Event` reference-counted so that we can access it after calling `schedulerpolicy_push()`.

Reply:
> Oh! I didn't realize that. I guess that's since moving the "cpu delay" code into the scheduler?

In this case, the `schedulerpolicy_push` may delay the event if `srcHost != dstHost` and the time is before the barrier (the end of the current round).